### PR TITLE
Clean channe/user/group name on post to

### PR DESCRIPTION
**Pull request by jesuslg123**
Added function to clean the user/channel/group name before try to find the destination on posTo message.

Function clean any # (from groups or channels) or @ (from users mentions) at the beginning of the name to allow find the destination when comparing by name.

Sample: Bot is mentioned on #general channel, if you try to reply using source message from message incoming, you will have "#general" instead of "general" making the search of destination on postTo, assert.

### CODE REVIEW

**Comment by mishk0** [comment]
Fine. Could you write few tests and add JSDoc?

**Comment by jesuslg123** [comment]
Sure, I will ASAP. Thank you!

**Comment by jesuslg123** [comment]
Done, I hope you like it :) @mishk0

File: index.js [hunk 1]
```diff
@@ -385,6 +387,23 @@ class Bot extends EventEmitter {
     }
 
     /**
+     * Remove @ or # character from group | channel | user name
+     * @param {string} name
+     * @returns {string}
+     */
+    _cleanName (name) {
+        if (name === undefined || name.lenght === 0) {
```
Comment (mishk0) [inline index.js]: Think, this condition is redundant.

File: index.js [hunk 2]
```diff
@@ -385,6 +387,23 @@ class Bot extends EventEmitter {
...
+            return name;
+        }
+
+        var firstCharacter = name[0];
```
Comment (mishk0) [inline index.js]: `charAt(0)`

File: index.js [hunk 3]
```diff
@@ -385,6 +387,23 @@ class Bot extends EventEmitter {
...
+        if (firstCharacter === '#' || firstCharacter === '@') {
+            name = name.substring(1, name.lenght);
```
Comment (mishk0) [inline index.js]: `name.slice(1)`

File: index.js (hunk 1 above)
Comment (mishk0) [inline index.js]: Probably, you can check that an argument is exist and it is a string `(name && typeof name === 'string')`

**Comment by jesuslg123** [comment]
@mishk0 I have reviewed and apply the changes already. Let me know if you have any other consideration. Thanks!

**Comment by mishk0** [comment]
Thanks, there is one more suggestion.

**Comment by jesuslg123** [comment]
Sorry, which one?

**Comment by jesuslg123** [comment]
Will you merge into master branch? @mishk0

**Comment by mishk0** [comment]
Sorry, forgot to send it.

File: index.js [hunk 4]
```diff
@@ -385,6 +387,23 @@ class Bot extends EventEmitter {
     }
 
     /**
+     * Remove @ or # character from group | channel | user name
+     * @param {string} name
+     * @returns {string}
+     */
+    _cleanName (name) {
+        if (typeof name === undefined || typeof name !== 'string') {
```
Comment (jesuslg123) [inline index.js]: You are totally right!

**Comment by jesuslg123** [comment]
Thank you! :) Just to know, will you update npm release version?

**Comment by mishk0** [comment]
i will ))

**Pull request merged**